homeassistant/components/icloud/__init__.py

```diff
@@ -1 +1,574 @@
-"""The icloud component."""
+"""The iCloud component."""
+from datetime import timedelta
+import logging
+import operator
+from typing import Dict
+
+from pyicloud import PyiCloudService
+from pyicloud.exceptions import PyiCloudFailedLoginException, PyiCloudNoDevicesException
+from pyicloud.services.findmyiphone import AppleDevice
+import voluptuous as vol
+
+from homeassistant.components.zone import async_active_zone
+from homeassistant.config_entries import SOURCE_IMPORT, ConfigEntry
+from homeassistant.const import ATTR_ATTRIBUTION, CONF_PASSWORD, CONF_USERNAME
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.dispatcher import dispatcher_send
+from homeassistant.helpers.event import track_point_in_utc_time
+from homeassistant.helpers.typing import ConfigType, HomeAssistantType
+from homeassistant.util import slugify
+from homeassistant.util.async_ import run_callback_threadsafe
+from homeassistant.util.dt import utcnow
+from homeassistant.util.location import distance
+
+from .const import (
+    CONF_ACCOUNT_NAME,
+    CONF_GPS_ACCURACY_THRESHOLD,
+    CONF_MAX_INTERVAL,
+    DEFAULT_GPS_ACCURACY_THRESHOLD,
+    DEFAULT_MAX_INTERVAL,
+    DOMAIN,
+    ICLOUD_COMPONENTS,
+    TRACKER_UPDATE,
+    DEVICE_BATTERY_LEVEL,
+    DEVICE_BATTERY_STATUS,
+    DEVICE_CLASS,
+    DEVICE_DISPLAY_NAME,
+    DEVICE_LOCATION,
+    DEVICE_LOCATION_LATITUDE,
+    DEVICE_LOCATION_LONGITUDE,
+    DEVICE_LOST_MODE_CAPABLE,
+    DEVICE_LOW_POWER_MODE,
+    DEVICE_NAME,
+    DEVICE_PERSON_ID,
+    DEVICE_STATUS,
+    DEVICE_STATUS_SET,
+    DEVICE_STATUS_CODES,
+)
+
+ATTRIBUTION = "Data provided by Apple iCloud"
+
+# entity attributes
+ATTR_ACCOUNT_FETCH_INTERVAL = "account_fetch_interval"
+ATTR_BATTERY = "battery"
+ATTR_BATTERY_STATUS = "battery_status"
+ATTR_DEVICE_NAME = "device_name"
+ATTR_DEVICE_STATUS = "device_status"
+ATTR_LOW_POWER_MODE = "low_power_mode"
+ATTR_OWNER_NAME = "owner_fullname"
+
+_LOGGER = logging.getLogger(__name__)
+
+SERVICE_ICLOUD_PLAY_SOUND = "play_sound"
+SERVICE_ICLOUD_DISPLAY_MESSAGE = "display_message"
+SERVICE_ICLOUD_LOST_DEVICE = "lost_device"
+SERVICE_ICLOUD_UPDATE = "update"
+SERVICE_ICLOUD_RESET = "reset"
+SERVICE_ATTR_USERNAME = CONF_USERNAME
```

Comment:
We call service data attribute constants `ATTR_*`. Please rename to `ATTR_USERNAME`.